x-pack/plugins/enterprise_search/server/lib/analytics/delete_analytics_collection.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IScopedClusterClient } from '@kbn/core-elasticsearch-server';
+
+import { ANALYTICS_COLLECTIONS_INDEX } from '../..';
+
+import { ErrorCode } from '../../../common/types/error_codes';
+import { fetchIndices } from '../indices/fetch_indices';
+
+import { fetchAnalyticsCollectionByName } from './fetch_analytics_collection';
+
+const deleteAnalyticsCollectionEvents = async (client: IScopedClusterClient, name: string) => {
+  const indexPattern = `elastic_analytics-events-${name}-*`;
+  const indices = await fetchIndices(client, indexPattern, true, false);
+
+  for (const index of indices) {
+    await client.asCurrentUser.indices.delete({
+      ignore_unavailable: true,
+      index: index.name,
+    });
+  }
+};
+
+export const deleteAnalyticsCollectionByName = async (
+  client: IScopedClusterClient,
+  name: string
+) => {
+  const analyticsCollection = await fetchAnalyticsCollectionByName(client, name);
```

Comment:
question: is this check necessary? I didn't check, but IIRC the delete API just throw an error if you try to delete a non-existent index. That would probably allow you to use the isIndexNotFoundError util in the top-level util we use elsewhere as well.

Reply:
I had this debate with myself. My reasoning is that as I have no rollback, I want to delete the events indices first before I delete the analytics collection document from the store. The difficulty then means if I delete events first, the API is trusting that the collection exists and allow to delete any indices with given name (although the indices name is prefixed by the API so the damage is small). I felt a quick cheap check to make sure it exists would be a better approach.

Reply:
Okay cool, makes sense to me!

Reply:
I would also need to use `delete_by_query` which I believe wont fail if the query doesn't match any docs to delete.